src/Microsoft.ML.OnnxTransformer/OnnxTypes.cs

```diff
@@ -0,0 +1,548 @@
+            return enumerableType;
+        }
+
+        /// <summary>
+        /// Create the corresponding <see cref="DataViewType"/> for ONNX sequence.
+        /// </summary>
+        /// <param name="elementType">The element type of a sequence.</param>
+        public OnnxSequenceType(Type elementType) : base(MakeNativeType(elementType))
+        {
+            DataViewTypeManager.Register(this, RawType, new[] { new OnnxSequenceTypeAttribute(elementType) });
+        }
+
+        public override bool Equals(DataViewType other)
+        {
+            if (other is OnnxSequenceType)
+                return RawType == other.RawType;
+            else
+                return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return RawType.GetHashCode();
+        }
+    }
+
+    /// <summary>
+    /// The corresponding <see cref="DataViewSchema.Column.Type"/> of ONNX's map type in <see cref="IDataView"/>'s type system.
+    /// In other words, if an ONNX model produces a map, a column in <see cref="IDataView"/> may be typed to <see cref="OnnxMapType"/>.
+    /// Its underlying type is <see cref="IDictionary{TKey, TValue}"/>, where the generic type "TKey" and "TValue" are the input arguments of
+    /// <see cref="OnnxMapType.OnnxMapType(Type,Type)"/>.
+    /// </summary>
+    public sealed class OnnxMapType : StructuredDataViewType
+    {
+        /// <summary>
+        /// Create the corresponding <see cref="DataViewType"/> for ONNX map.
+        /// </summary>
+        /// <param name="keyType">Key type of the associated ONNX map.</param>
+        /// <param name="valueType">Value type of the associated ONNX map.</param>
+        public OnnxMapType(Type keyType, Type valueType) : base(typeof(IDictionary<,>).MakeGenericType(keyType, valueType))
+        {
+            DataViewTypeManager.Register(this, RawType, new[] { new OnnxMapTypeAttribute(keyType, valueType) });
+        }
+
+        public override bool Equals(DataViewType other)
+        {
+            if (other is OnnxMapType)
+                return RawType == other.RawType;
+            else
+                return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return RawType.GetHashCode();
+        }
+    }
+
+    /// <summary>
+    /// To declare <see cref="OnnxSequenceType"/> column in <see cref="IDataView"/> as a field
+    /// in a <see langword="class"/>, the associated field should be marked with <see cref="OnnxSequenceTypeAttribute"/>.
+    /// Its uses are similar to those of <see cref="VectorTypeAttribute"/> and other <see langword="class"/>es derived
+    /// from <see cref="DataViewTypeAttribute"/>.
+    /// </summary>
+    public sealed class OnnxSequenceTypeAttribute : DataViewTypeAttribute
+    {
+        private Type _elemType;
+        /// <summary>
+        /// Create an image type without knowing its height and width.
```

Comment:
Copy and paste left over.  Please fix everywhere.

Reply:
You're right. Many thanks.